cmd/api.go

```diff
@@ -9,18 +9,40 @@ import (
 	"net/url"
 
 	"github.com/dyninc/qstring"
+	"github.com/go-playground/validator/v10"
 )
 
+type Period string
+
+const (
+	OVERALL       Period = "overall"
+	SEVEN_DAYS    Period = "7day"
+	ONE_MONTH     Period = "1month"
+	THREE_MONTHS  Period = "3month"
+	SIX_MONTHS    Period = "6month"
+	TWELVE_MONTHS Period = "12month"
+)
+
+func validatePeriod(fl validator.FieldLevel) bool {
+	period := Period(fl.Field().String())
+	switch period {
+	case OVERALL, SEVEN_DAYS, ONE_MONTH, THREE_MONTHS, SIX_MONTHS, TWELVE_MONTHS:
+		return true
+	default:
+		return false
+	}
+}
+
 type CollageRequest struct {
-	Rows     int    `url:"rows"`
-	Columns  int    `url:"columns"`
-	Username string `url:"username"`
-	Period   string `url:"period"`
+	Rows     int    `url:"rows" validate:"required,gte=1,lte=10"`
+	Columns  int    `url:"columns" validate:"required,gte=1,lte=10"`
+	Username string `url:"username" validate:"required"`
+	Period   Period `url:"period" validate:"required,validatePeriod"`
 }
 
-func get_collage(request *CollageRequest) image.Image {
-	limit := request.Rows * request.Columns
-	albums := get_albums(request.Username, request.Period, limit)
+func getCollage(request *CollageRequest) image.Image {
+	count := request.Rows * request.Columns
+	albums := get_albums(request.Username, request.Period, count)
```

Comment:
I think should be `getAlbums`

Reply:
Same on line `52` with `create_collage`

Reply:
Good catch! I ran the rename but the `api.go` file didn't get saved lol